Add page around PR checks to contribution guidelines. (#4006)
open-telemetry/opentelemetry.io: content/en/docs/contributing/pr-checks.md

---
title: Pull request checks
description: Learn how to make your pull request successfully pass all checks
weight: 40
cSpell:ignore: REFCACHE
---

When you raise a
[pull request](https://docs.github.com/en/get-started/learning-about-github/github-glossary#pull-request)
(PR) with the
[opentelemetry.io repository](https://github.com/open-telemetry/opentelemetry.io)
a set of checks are executed. The PR checks verify that...

- … you have signed the [CLA](#easy-cla).
- …your commit can be deployed through [Netlify](#netlify-deployment)
  successfully.
- … your changes are compliant with our [style guide](#style-checks).

{{% alert title="Note" color="primary" %}}

If any of the PR checks fails, please try to
[fix content issues automatically](/docs/contributing/#fix-content-issues-automatically)

> first by running `npm run test-and-fix` on your machine.

Author: ```suggestion
first by running `npm run fix:all` on your machine.
```

If any of the PR checks fails, try to
[fix content issues automatically](/docs/contributing/#fix-content-issues-automatically)
by running `npm run test-and-fix` on your machine.

Additionally, you can comment `/fix:all` on your Pull Request. This will make
the OpenTelemetry Bot run those commands on your behalf and update the PR. Make
sure that you pull those changes locally.

{{% /alert %}}

## Easy CLA

This check fails if you haven't
[signed the CLA](/docs/contributing/#sign-the-cla).

## Netlify deployment

If the [Netlify](https://www.netlify.com/) build fails, select **Details** for
more information.

## Style checks

To make sure that contributions follow our
[style guide](/docs/contributing/style-guide) we have implemented a set of
checks that verify style guide rules and fail l if they find any issues.

The following list describes current checks and what you can do to fix related
errors:

- `TEXT linter`: This check verifies that
  [OpenTelemetry-specific terms and words are used consistently across the site](/docs/contributing/style-guide#opentelemetryio-word-list).
  If any issues are found, annotations are added to your files in the

>   `files changed` view of your PR. Fix those to turn the check green. As an
>   alternative, you can run `npm run check:text` locally and fix all issues
>   highlighted in the output.

Author: ```suggestion
  `files changed` view of your PR. Fix those to turn the check green. As an
  alternative, you can run `npm run check:text -- --fix` locally to fix most issues.
  Run `npm run check:text` again and manually fix the remaining issues.
```

  [all words are spelled correctly](/docs/contributing/style-guide#spell-checking).
- `CSPELL:IGNORE check`: This check will verify that all words in your cSpell
  ignore list are normalized. If this check fails, run `npm run fix:dict`
  locally and push the changes in a new commit.
- `FILENAME check`: This check verifies that all
  [files are formatted by prettier](/docs/contributing/style-guide#file-format).
  If this check fails, run `npm fix:format` locally and push the changes in a
  new commit.
- `FILE FORMAT`: This check verifies that all
  [file names are in kebab-case](/docs/contributing/style-guide#file-names). If
  this check fails, run `npm fix:filenames` locally and push the changes in a
  new commit.
- `BUILD and CHECK LINKS` / `REFCACHE updates?`: This check verifies that all
  links that your commits are introducing are functional. Run
  `npm run check:links` to check them locally. This also updates the reference
  cache, or `REFCACHE`. Push any changes to the `REFCACHE` in a new commit.
- `WARNINGS in build log?`: If this check fails, review the build log for any
  other potential issues. Ask maintainers for help, if you are unsure how to

>   recover.

Reviewer: I would suggest putting each entry into its own H3 section. We could then put the commands in code block, which would make them easier to spot.

That said, this is ok to merge as is IMHO.

Author: fixed

---

> title: Pull Request Checks

Reviewer 2: ```suggestion
title: Pull request checks
```

> description: Learn how to make your Pull Request successfully pass all checks

Reviewer 2: ```suggestion
description: Learn how to make your pull request successfully pass all checks
```

weight: 40
cSpell:ignore: REFCACHE
---

> When you raise a
> [Pull Request](https://docs.github.com/en/get-started/learning-about-github/github-glossary#pull-request)
> with the
> [opentelemetry.io repository](https://github.com/open-telemetry/opentelemetry.io)
> a set of checks will be run. Those checks will verify, …
> 
> - … if you have signed the [CLA](#easy-cla).
> - …if your commit can be deployed via [Netlify](#netlify-deployment)
>   successfully.
> - … if your changes are in line with some [style guidelines](#style-checks).

Reviewer 2: ```suggestion
When you raise a
[pull request](https://docs.github.com/en/get-started/learning-about-github/github-glossary#pull-request)
(PR) with the
[opentelemetry.io repository](https://github.com/open-telemetry/opentelemetry.io)
a set of checks are executed. The PR checks verify that...

- … you have signed the [CLA](#easy-cla).
- …your commit can be deployed through [Netlify](#netlify-deployment)
  successfully.
- … your changes are compliant with our [style guide](#style-checks).
```

GitHub doesn't capitalize "pull request", though the initialism is PR.

{{% alert title="Note" color="primary" %}}

> If any of your PR checks fails, try to
> [fix content issues automatically](/docs/contributing/#fix-content-issues-automatically)
> first by running `npm run test-and-fix` locally.
> 
> Additionally, you can comment `/fix:all` on your Pull Request. This will make
> the OpenTelemetry Bot run those commands on your behalf and update your PR. Make
> sure that you pull those changes locally.

Reviewer 2: ```suggestion
If any of the PR checks fails, try to
[fix content issues automatically](/docs/contributing/#fix-content-issues-automatically)
by running `npm run test-and-fix` on your machine.

Additionally, you can comment `/fix:all` on your Pull Request. This will make
the OpenTelemetry Bot run those commands on your behalf and update the PR. Make
sure that you pull those changes locally.
```

{{% /alert %}}

## Easy CLA

> This check will fail, if you didn't
> [sign the CLA](/docs/contributing/#sign-the-cla).

Reviewer 2: ```suggestion
This check will fail, if you didn't
[sign the CLA](/docs/contributing/#sign-the-cla).
```
```suggestion
This check fails if you haven't
[signed the CLA](/docs/contributing/#sign-the-cla).
```

Reviewer 2: Can we add a link to the CLA instructions?

Author: This links back to the contributing index page where it is outlined what needs to be done to sign the CLA

> ## Netlify Deployment

Reviewer 2: ```suggestion
## Netlify deployment
```

If the [Netlify](https://www.netlify.com/) build fails, select **Details** for
more information.

## Style checks

To make sure that contributions follow our
[style guide](/docs/contributing/style-guide) we have implemented a set of

> checks that will verify certain rules of that style guide and will fail if they

Reviewer 2: ```suggestion
checks that verify style guide rules and fail l if they
```

find any issues.

> Below you will find a list of the checks that are run and what you can do to fix
> any errors:

Reviewer 2: ```suggestion
The following list describes current checks and what you can do to fix
related errors:
```

> - `TEXT linter`: This check will verify that
>   [OpenTelemetry-specific terms and words are used consistently across the site](/docs/contributing/style-guide#opentelemetryio-word-list).
>   If any issues are found, annotations will be added to your files in the
>   `files changed` view of your PR. Fix those to turn the check green. As an
>   alternative you can run `npm run check:text` locally and fix all issues
>   highlighted in the output.
> - `MARKDOWN linter`: This check will verify that
>   [standards and consistency for Markdown files are enforced](/docs/contributing/style-guide#markdown-standards).
>   If any issues are found, run `npm:run format` to fix (most) issues. For more
>   complex issues, run `npm run check:markdown` and apply the suggested changes.
> - `SPELLING check`: This check will verify that
>   [all words are spelled correctly](/docs/contributing/style-guide#spell-checking).
> - `CSPELL:IGNORE check`: This check will verify that all words in your cSpell
>   ignore list are normalized. If this check fails, run `npm run fix:dict`
>   locally and push the changes in a new commit.
> - `FILENAME check`: This check will verify that all
>   [files are formatted by prettier](/docs/contributing/style-guide#file-format).
>   If this check fails, run `npm fix:format` locally and push the changes in a
>   new commit.
> - `FILE FORMAT`: This check will verify that all
>   [file names are in kebab-case](/docs/contributing/style-guide#file-names). If
>   this check fails, run `npm fix:filenames` locally and push the changes in a
>   new commit.
> - `BUILD and CHECK LINKS` / `REFCACHE updates?`: This check will verify that all
>   links that your commits are introducing are functional. Run
>   `npm run check:links` to check them locally. This will also update the so
>   called `REFCACHE`. Push any changes to the `REFCACHE` in a new commit.
> - `WARNINGS in build log?`: If this check fails, review the build log for any
>   other potential issues. Ask maintainers for help, if you are unsure how to
>   recover.

Reviewer 2: ```suggestion
- `TEXT linter`: This check verifies that
  [OpenTelemetry-specific terms and words are used consistently across the site](/docs/contributing/style-guide#opentelemetryio-word-list).
  If any issues are found, annotations are added to your files in the
  `files changed` view of your PR. Fix those to turn the check green. As an
  alternative, you can run `npm run check:text` locally and fix all issues
  highlighted in the output.
- `MARKDOWN linter`: This check verifies that
  [standards and consistency for Markdown files are enforced](/docs/contributing/style-guide#markdown-standards).
  If any issues are found, run `npm:run format` to fix most issues. For more
  complex issues, run `npm run check:markdown` and apply the suggested changes.
- `SPELLING check`: This check verifies that
  [all words are spelled correctly](/docs/contributing/style-guide#spell-checking).
- `CSPELL:IGNORE check`: This check will verify that all words in your cSpell
  ignore list are normalized. If this check fails, run `npm run fix:dict`
  locally and push the changes in a new commit.
- `FILENAME check`: This check verifies that all
  [files are formatted by prettier](/docs/contributing/style-guide#file-format).
  If this check fails, run `npm fix:format` locally and push the changes in a
  new commit.
- `FILE FORMAT`: This check verifies that all
  [file names are in kebab-case](/docs/contributing/style-guide#file-names). If
  this check fails, run `npm fix:filenames` locally and push the changes in a
  new commit.
- `BUILD and CHECK LINKS` / `REFCACHE updates?`: This check verifies that all
  links that your commits are introducing are functional. Run
  `npm run check:links` to check them locally. This also updates the reference cache,
  or `REFCACHE`. Push any changes to the `REFCACHE` in a new commit.
- `WARNINGS in build log?`: If this check fails, review the build log for any
  other potential issues. Ask maintainers for help, if you are unsure how to
  recover.
```
